src/quartz_api/internal/inputs/dataplatform/client.py

```diff
@@ -0,0 +1,277 @@
+"""A data platform implementation that conforms to the DatabaseInterface."""
+
+from typing import override
+
+from dp_sdk.ocf import dp
+from fastapi import HTTPException
+
+from quartz_api import internal
+from quartz_api.internal.models import ForecastHorizon
+
+from ..utils import get_window
+
+
+class Client(internal.DatabaseInterface):
+    """Defines a data platform interface that conforms to the DatabaseInterface."""
+
+    c: dp.DataPlatformDataServiceStub
+
+    @classmethod
+    def new(cls, dp_client: dp.DataPlatformDataServiceStub) -> "Client":
+        """Class method to create a new Data Platform client."""
+        instance = cls()
+        instance.c = dp_client
+        return instance
+
+    @override
+    async def get_predicted_solar_power_production_for_location(
+        self,
+        location: str,
+        forecast_horizon: ForecastHorizon = ForecastHorizon.latest,
+        forecast_horizon_minutes: int | None = None,
+        smooth_flag: bool = True,
+    ) -> list[internal.PredictedPower]:
+        values = await self._get_predicted_power_production_for_location(
+            location,
+            dp.EnergySource.SOLAR,
+            forecast_horizon,
+            forecast_horizon_minutes,
+            smooth_flag,
+        )
+        return values
+
+    @override
+    async def get_predicted_wind_power_production_for_location(
+        self,
+        location: str,
+        forecast_horizon: ForecastHorizon = ForecastHorizon.latest,
+        forecast_horizon_minutes: int | None = None,
+        smooth_flag: bool = True,
+    ) -> list[internal.PredictedPower]:
+        values = self._get_predicted_power_production_for_location(
+            location,
+            dp.EnergySource.WIND,
+            forecast_horizon,
+            forecast_horizon_minutes,
+            smooth_flag,
+        )
+        return values
+
+    @override
+    async def get_actual_solar_power_production_for_location(
+        self,
+        location: str,
+    ) -> list[internal.ActualPower]:
+        return self._get_actual_power_production_for_location(
+            location,
+            dp.EnergySource.SOLAR,
+        )
+
+    @override
+    async def get_actual_wind_power_production_for_location(
+        self,
+        location: str,
+    ) -> list[internal.ActualPower]:
+        return self._get_actual_power_production_for_location(
+            location,
+            dp.EnergySource.WIND,
+        )
+
+    @override
+    async def get_wind_regions(self) -> list[str]:
+        req = dp.ListLocationsRequest(
+            energy_source_filter=dp.EnergySource.WIND,
+            location_type_filter=dp.LocationType.SITE,
```

Comment:
Should you leave a TODO in there, slightly concious that you are asking for regions, but we are getting sites

Reply:
Updated